python/pyarrow/tests/test_pandas.py

```diff
@@ -2882,6 +2882,50 @@ def test_variable_dictionary_with_pandas():
         a.to_pandas()
 
 
+# ----------------------------------------------------------------------
+# Array protocol in pandas conversions tests
+
+
+def test_array_protocol():
+
+    def __arrow_array__(self, type=None):
+        return pa.array(self._data, mask=self._mask, type=type)
+
+    df = pd.DataFrame({'a': pd.Series([1, 2, None], dtype='Int64')})
```

Comment:
"Int64" (capitalized) is a Pandas-specific type name?

Reply:
Yes, that's the (rather subtle) way to distinguish between default numpy `'int64'` (in string form)